pkg/watchdog/pkg_test.go

```diff
@@ -4,8 +4,374 @@
 
 package watchdog
 
-import "testing"
+import (
+	"fmt"
+	"io/ioutil"
+	"testing"
+	"time"
 
-func TestTODO(t *testing.T) {
-	// TODO: Write a unit test.
+	"golang.org/x/sys/unix"
+)
+
+func getTestStruct() (*Watchdog, error) {
+	f, err := ioutil.TempFile("", "testfile-")
+	if err != nil {
+		return nil, err
+	}
+	return &Watchdog{
+		File:               f,
+		unixSyscall:        nil,
+		unixIoctlGetUint32: nil,
+	}, nil
+}
+
+func TestOpen(t *testing.T) {
+	f, err := ioutil.TempFile("", "testfile-")
+	if err != nil {
+		t.Error(err)
+	}
+	defer f.Close()
+
+	w, err := Open(f.Name())
+	if err != nil {
+		t.Error(err)
+	}
+	defer w.Close()
+}
+
+func TestOpenError(t *testing.T) {
+	_, err := Open("testfile")
```

Comment:
if _, err := Open(); err == nil {
etc.
Also be sure to look at Ryan's PR on testing rules, we need more formal test messages.
}

Reply:
Be a bit more explicit with the file name like this:

```suggestion
	_, err := Open("file-that-does-not-exist")
```

Reply:
Removed it as well, because we dont test the Open function. (See above)